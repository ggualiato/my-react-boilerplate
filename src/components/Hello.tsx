import React, { FC } from "react";

interface IHelloProps {
    text: string;
    onClick: () => void;
}

export const Hello: FC<IHelloProps> = (props) => {
    return <button onClick={props.onClick}>{props.text}</button>;
};
