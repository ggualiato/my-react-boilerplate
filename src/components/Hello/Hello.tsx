import React, { FC } from "react";

interface HelloProps {
    text: string;
    onClick: () => void;
}

export const Hello: FC<HelloProps> = (props) => {
    return <button onClick={props.onClick}>{props.text}</button>;
};
