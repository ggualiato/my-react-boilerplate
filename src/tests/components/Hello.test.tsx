import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Hello from "../../components/Hello/Hello";

test("should render Hello correctly", () => {
    const onClick = jest.fn();
    const { container, getByRole } = render(<Hello text="Test" onClick={onClick} />);
    const button = getByRole("button");

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
});
