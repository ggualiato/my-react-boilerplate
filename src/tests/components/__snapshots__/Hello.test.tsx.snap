// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Hello correctly 1`] = `
<button
  onClick={[Function]}
>
  Test
</button>
`;
